### PR TITLE
Binance: Implicit API Options Examples

### LINKED ISSUE #15406: Help with Binance European style-vanilla options
- OS: Windows 10 (21H2)
- Programming Language version: node.js (ver 18.10.0)
- CCXT version: 2.0.37

# Question

Does ccxt offer possibility to trade european options on binance? On mobile application i can select call, put and vol orders and time frame from 5min to 1day expiration. Supported coins are BTCUSDT, ETHUSDT ,XRPUSDT, LTCUSDT, LINKUSDT and BNBUSDT.

Also, if it is possible i can not find any working example.

Thank you.

# Attached picture from mobile...

0b846bc180b14984b10a3e411aa7bdc3

### PR DESCRIPTION
**Pull request by Dan-krm**
Created async implicit API option trading examples for binance in python:
fixes #15406

### examples/py/async-binance-create-option-order.py
```
{
    'mmp': False,
    'orderId': '4612100534317743431',
    'postOnly': False,
    'price': '2.1',
    'quantity': '1',
    'reduceOnly': False,
    'side': 'BUY',
    'symbol': 'ETH-221028-1700-C',
    'type': 'LIMIT'
}
```

### examples/py/async-binance-fetch-option-order.py
```
[
    {
        'avgPrice': '0.00000000',
        'clientOrderId': '',
        'createTime': '1666826857206',
        'executedQty': '0.00000000',
        'fee': '0.00000000',
        'mmp': False,
        'optionSide': 'CALL',
        'orderId': '4612098335294532880',
        'postOnly': False,
        'price': '2.10000000',
        'priceScale': '1',
        'quantity': '1.00000000',
        'quantityScale': '2',
        'quoteAsset': 'USDT',
        'reduceOnly': False,
        'side': 'BUY',
        'source': 'API',
        'status': 'ACCEPTED',
        'symbol': 'ETH-221028-1700-C',
        'timeInForce': 'GTC',
        'type': 'LIMIT',
        'updateTime': '1666826857206'
    }
]
```

### examples/py/async-binance-cancel-option-order.py
```
{
    'avgPrice': '0',
    'clientOrderId': '',
    'createTime': '1666826059423',
    'executedQty': '0.00',
    'fee': '0',
    'mmp': False,
    'optionSide': 'CALL',
    'orderId': '4612100534317768959',
    'postOnly': False,
    'price': '2.1',
    'priceScale': '1',
    'quantity': '1.00',
    'quantityScale': '2',
    'quoteAsset': 'USDT',
    'reduceOnly': False,
    'side': 'BUY',
    'source': 'API',
    'status': 'CANCELLED',
    'symbol': 'ETH-221028-1700-C',
    'timeInForce': 'GTC',
    'type': 'LIMIT',
    'updateTime': '1666826400726'
}
```

### examples/py/async-binance-fetch-option-position.py
```
[
    {
        'entryPrice': '1.50000000',
        'expiryDate': '1666944000000',
        'markPrice': '1.1',
        'markValue': '1.100000000',
        'optionSide': 'CALL',
        'positionCost': '1.50000000',
        'priceScale': '1',
        'quantity': '1.00000000',
        'quantityScale': '2',
        'quoteAsset': 'USDT',
        'reducibleQty': '1.00000000',
        'ror': '-0.2666',
        'side': 'LONG',
        'strikePrice': '1800.00000000',
        'symbol': 'ETH-221028-1800-C',
        'time': '1666827652709',
        'unrealizedPNL': '-0.400000000'
    }
]
```

### examples/py/async-binance-fetch-option-details.py
```
[
    {
        'askIV': '0.96028566',
        'bidIV': '0.84410369',
        'delta': '0.08242491',
        'gamma': '0.00174896',
        'highPriceLimit': '291.50',
        'lowPriceLimit': '0.10',
        'markIV': '0.90219468',
        'markPrice': '3.1',
        'symbol': 'ETH-221028-1700-C',
        'theta': '-4.81838804',
        'vega': '0.14652545'
    },
    ...
]
```

### examples/py/async-binance-fetch-option-ticker.py
```
[
    {
        'amount': '263.61',
        'askPrice': '0.2',
        'bidPrice': '0',
        'closeTime': '1666917419604',
        'exercisePrice': '1508.3269565',
        'firstTradeId': '157',
        'high': '5',
        'lastPrice': '0.2',
        'lastQty': '1.5',
        'low': '0.2',
        'open': '3.1',
        'openTime': '1666846538303',
        'priceChange': '-2.9',
        'priceChangePercent': '-0.9354',
        'strikePrice': '1700',
        'symbol': 'ETH-221028-1700-C',
        'tradeCount': '50',
        'volume': '213.9'
    }
]
```

### examples/py/async-binance-fetch-option-OHLCV.py
```
[
    {
        'amount': '0',
        'close': '0.2',
        'closeTime': '1666930800000',
        'high': '0.2',
        'interval': '1m',
        'low': '0.2',
        'open': '0.2',
        'openTime': '1666930740000',
        'takerAmount': '0',
        'takerVolume': '0',
        'tradeCount': '0',
        'volume': '0'
    },
    ...
]
```

### examples/py/async-binance-fetch-option-orderbook.py
```
{
    'T': '1666930433849',
    'asks': [
        ['14.9', '0.1'],
        ['17.6', '0.01'],
        ['26.9', '0.9']
        ...
    ],
    'bids': [
        ['2.3', '0.1'],
        ['2.1', '1'],
        ['1', '2']
        ...
    ]
}
```

### CODE REVIEW

File: examples/py/async-binance-cancel-option-order.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+
```
Comment (carlosmiei) [inline examples/py/async-binance-cancel-option-order.py:2]: @Dan-krm thanks for creating these examples, can you please create one additional file with some public calls? (orderbook, ticker, etc)

File: examples/py/async-binance-create-option-order.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import os
+import sys
+from pprint import pprint
+
+root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+sys.path.append(root + '/python')
+
+import ccxt.async_support as ccxt  # noqa: E402
+
+
```
Comment (carlosmiei) [inline examples/py/async-binance-create-option-order.py:13]: We could probably add a comment to each file saying something like this:

```
This example uses the implicit API, so this is not the most ergonomic way to handle it, in the future we will have options properly unified which will make things much easier.
You can check if they're already ready-to-use with the unified methods  (createOrder, fetchOrder, fetchTicker, etc) by doing `is_unified = exchange.has['option']`
```

wdyt?

File: examples/py/async-binance-cancel-option-order.py (hunk 1 above)
Comment (Dan-krm) [reply examples/py/async-binance-cancel-option-order.py:2]: Yeah I'll add some more public examples

File: examples/py/async-binance-create-option-order.py (hunk 2 above)
Comment (Dan-krm) [reply examples/py/async-binance-create-option-order.py:13]: Yeah I think that's a good idea, I'll add a comment to each file

**Comment by carlosmiei** [comment]
LGTM ✅

**Review by carlosmiei — APPROVED** [review]

**Pull request merged**